packages/app-project/stores/User/UserPersonalization/YourStats/YourStats.js

```diff
@@ -74,7 +80,7 @@ const YourStats = types
         weekDay.setUTCDate(newDate)
         const period = weekDay.toISOString().substring(0, 10)
         const { count } = dailyCounts.find(count => count.period.startsWith(period)) || { count: 0, period }
-        const dayNumber = weekDay.getDay()
+        const dayNumber = weekDay.getUTCDay()
```

Comment:
I think this was causing the weekday bug in #2244, but it only triggers when you run `getDay()` in a time zone other than UTC.

Reply:
See #2528 for some context on why the code is getting the weekday here.